hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoGetAccountInfoHandlerTest.java

```diff
@@ -238,19 +238,19 @@ void getsResponseIfOkResponse() {
 
         given(token1.decimals()).willReturn(100);
         given(token1.symbol()).willReturn("FOO");
-        given(token1.tokenNumber()).willReturn(3L);
+        given(token1.tokenId()).willReturn(asToken(3L));
```

Comment:
Might be a future PR, but it would be good to have a few of these test scenarios use token ID values with non-zero shard and realm.

Reply:
Agreed. Currently lots of code use default shard, realm 0 value.

Reply:
Almost certainly a future PR, but please put in an issue for that.  We really do need to start adding unit tests that break if we make assumptions about shard and realm, so we can find and remove those assumptions.

Reply:
We also need to add shard and realm to the relevant pbj objects. Some of them don't even have shard or realm

Reply:
We've been adding them, that's the primary purpose of this PR, adding ID instead of number so we retain shard and realm for all the items in state.